File: sdk/python/apex_sdk/intelligence.py
"""Python SDK for the intelligence service."""

from __future__ import annotations

from typing import Any, Dict, List
from .client import APEXClient


class IntelligenceAPI:
    def __init__(self, client: APEXClient) -> None:
        self._client = client

    async def ingest(self, space_id: str, content: str) -> str:
        """Ingest a document into a memory space.

        Returns the document ID.
        """
        payload = {"space_id": space_id, "content": content}
        res = await self._client.post(self._client.intelligence_url, "/memory/ingest", payload)
        res.raise_for_status()
        return res.json()["doc_id"]

    async def query(self, space_id: str, query: str, top_k: int = 5) -> List[Dict[str, Any]]:
        payload = {"space_id": space_id, "query": query, "top_k": top_k}
        res = await self._client.post(self._client.intelligence_url, "/memory/query", payload)
        res.raise_for_status()
        return res.json()["results"]

    async def completion(self, prompt: str, max_tokens: int = 256) -> str:
        payload = {"prompt": prompt, "max_tokens": max_tokens}
        res = await self._client.post(self._client.intelligence_url, "/llm/completions", payload)
        res.raise_for_status()
        return res.json().get("completion") or res.json().get("completion", "")

File: sdk/typescript/src/intelligence.ts
export async function ingest(spaceId: string, content: string, baseUrl?: string): Promise<string> {
  const url = `${baseUrl || process.env.NEXT_PUBLIC_INTELLIGENCE_URL || ''}/memory/ingest`;
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ space_id: spaceId, content })
  });
  const data = await res.json();
  if (!res.ok) throw new Error(data.detail || 'Failed to ingest');
  return data.doc_id;
}

export async function query(spaceId: string, queryText: string, topK = 5, baseUrl?: string) {
  const url = `${baseUrl || process.env.NEXT_PUBLIC_INTELLIGENCE_URL || ''}/memory/query`;
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ space_id: spaceId, query: queryText, top_k: topK })
  });
  const data = await res.json();
  if (!res.ok) throw new Error(data.detail || 'Failed to query');
  return data.results;
}

export async function completion(prompt: string, maxTokens = 256, baseUrl?: string) {
  const url = `${baseUrl || process.env.NEXT_PUBLIC_INTELLIGENCE_URL || ''}/llm/completions`;
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ prompt, max_tokens: maxTokens })
  });
  const data = await res.json();
  if (!res.ok) throw new Error(data.detail || 'Failed to complete');
  return data.completion || data;
}